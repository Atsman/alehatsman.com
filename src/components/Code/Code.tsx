import React, { FC } from 'react'
import Highlight, { defaultProps, Language } from 'prism-react-renderer'
import github from 'prism-react-renderer/themes/github'

import { Box } from '@/components/Box'

interface Props {
  className: string
}

export const Code: FC<Props> = ({ children, className }) => {
  const language = className ? className.replace(/language-/, '') : ''
  return (
    <Highlight
      {...defaultProps}
      theme={github}
      code={children.toString().trim()}
      language={language as Language}
    >
      {({ className, style, tokens, getLineProps, getTokenProps }) => (
        <Box padding={3} className={className} style={{ ...style, overflow: 'scroll' }}>
          {tokens.map((line, i) => (
            <div key={i} {...getLineProps({ line, key: i })}>
              {line.map((token, key) => (
                <span key={key} {...getTokenProps({ token, key })} />
              ))}
            </div>
          ))}
        </Box>
      )}
    </Highlight>
  )
}
