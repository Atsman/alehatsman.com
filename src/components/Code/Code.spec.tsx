import React from 'react'
import { render } from '@testing-library/react'
import { Code } from './Code'

describe('Code', () => {
  it('should match snapshot', () => {
    const fragment = render(
      <Code className="language-javascript">
        {'console.log(\'hello world\')'}
      </Code>
    ).asFragment()

    expect(fragment).toMatchSnapshot()
  })
})
