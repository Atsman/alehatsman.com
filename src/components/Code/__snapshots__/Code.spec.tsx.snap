// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code should match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  padding: 16px;
}

<div
    class="prism-code language-javascript emotion-0 emotion-1"
    style="color: rgb(36, 41, 46); background-color: rgb(246, 248, 250); overflow: auto;"
  >
    <div
      class="token-line"
      style="color: rgb(36, 41, 46);"
    >
      <span
        class="token console class-name"
        style="color: rgb(227, 98, 9);"
      >
        console
      </span>
      <span
        class="token punctuation"
        style="color: rgb(36, 41, 46);"
      >
        .
      </span>
      <span
        class="token method function property-access"
        style="color: rgb(111, 66, 193);"
      >
        log
      </span>
      <span
        class="token punctuation"
        style="color: rgb(36, 41, 46);"
      >
        (
      </span>
      <span
        class="token string"
      >
        'hello world'
      </span>
      <span
        class="token punctuation"
        style="color: rgb(36, 41, 46);"
      >
        )
      </span>
    </div>
  </div>
</DocumentFragment>
`;
